sys/include/shell.h

```diff
@@ -75,6 +75,15 @@ extern "C" {
 
 /**
  * @brief Default shell buffer size (maximum line length shell can handle)
+ *
+ * @warning On terminals that buffer input and send the full command line in one go,
+ *   it may be necessary to increase the @ref STDIO_RX_BUFSIZE to make
+ *   practical use of this buffer, especially because the current mechanism of
+ *   passing stdin (`isrpipe_t stdin_isrpipe`) does not support backpressure
+ *   and overflows silently. As a consequence, commands through such terminals
+ *   appear to be truncated at @ref STDIO_RX_BUFSIZE bytes (defaulting to 64)
+ *   unless the command is sent in parts (on many terminals, by presing Ctrl-D
+ *   half way through the command).
  */
 #define SHELL_DEFAULT_BUFSIZE   (128)
```

Comment:
Why is this not the same as `STDIO_RX_BUFSIZE`?

Reply:
I can't comment on the authors' intentions, but generally I do still think that these are different -- SHELL_DEFAULT_BUFSIZE is about the longest command, STDIO_RX_BUFSIZE is about how many bytes are tolerated to be sent w/o blocking on the application to process them into the buffer (and 64 is surprisingly large for that purpose). Their relation only matters in the case of a line-buffered terminal, which is a choice outside of the control of the embedded application.

If we want to change code and not just docs, I'd prefer we do it right (by allowing backpressure or doing zero-copy) rather than creating a cross-layer configuration dependency when we're already struggling to get our configs transparent.